### PR TITLE
cmd/server: download files if env vars are populated

### PR DESCRIPTION
**Pull request by adamdecaf**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,15 @@ The Fed project implements an HTTP server and [Go library](https://pkg.go.dev/gi
 
 Moov Fed can read the data files from anywhere on the filesystem. This allows you to mount the files and set `FEDACH_DATA_PATH` / `FEDWIRE_DATA_PATH` environmental variables. Both official formats from the Federal Reserve (plaintext and JSON) are supported.
 
+#### Download Files
+
+The Federal Reserve Board (FRB) eServices offers API access to download the files. To download these work with your ODFI / banking partner to obtain a download code. Then run Fed with the following environmental variables set.
```
Comment (vxio) [inline README.md:64]:
```suggestion
The Federal Reserve Board (FRB) eServices offers API access to download the files. To download these files, work with your ODFI / banking partner to obtain a download code. Then run Fed with the following environment variables set.
```

File: cmd/server/main.go [hunk 2]
```diff
@@ -148,15 +149,20 @@ func addPingRoute(r *mux.Router) {
 	})
 }
 
-func setupSearcher(logger log.Logger, s *searcher, achPath, wirePath string) error {
-	logger.Logf("search: loading %s for ACH data", achPath)
-	if err := s.readFEDACHData(achPath); err != nil {
-		return fmt.Errorf("error reading ACH file at %s: %v", achPath, err)
+func setupSearcher(logger log.Logger, s *searcher, achFile, wireFile *os.File) error {
+	if achFile == nil || wireFile == nil {
+		return errors.New("missing fedach or fedwire data file(s)")
```
Comment (vxio) [inline cmd/server/main.go:154]: I recommend breaking these into separate if-statements for a better error msg to the user.

Comment (adamdecaf) [reply cmd/server/main.go]: Yep, good call.

File: pkg/download/download.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	if opts.HTTPClient == nil {
+		opts.HTTPClient = &http.Client{
+			// These files can be fairly large to buffer to us and the FRB
+			// services can be slow, so we default to a hefty timeout.
+			Timeout: 90 * time.Second,
+		}
+	}
+
+	if opts.RoutingNumber == "" || opts.DownloadCode == "" {
+		return nil, errors.New("missing routing number or download code")
```
Comment (vxio) [inline pkg/download/download.go:43]: Oops, I missed this one, but I'd recommend splitting these too.

**Review by vxio — APPROVED** [review]

**Review by vxio — APPROVED** [review]

**Pull request merged**